### DISCUSSION TITLE
High availability using IKEv2

### DISCUSSION THREAD

**Discussion by avachette** [strongswan/strongswan#2480] [Ideas]
Hi everyone,

Just wanted to say a word on how strongswan is a great piece of software first of all.

I have a question about HA setup scenario described here:
- https://docs.strongswan.org/docs/5.9/features/highAvailability.html#_clusterip

It doesn't seem to work anymore on a vast majority of GNU/Linux distribution either using IPtables or nftables.

I tried to switch from CLUSTERIP to CLUSTER module but it's not working at all when I setup netfilter rules.

Some years ago I used a lot OpenBSD with the combo of CARP (VRRP's implementation) and sasyncd (synchronizes IPsec SA and SPD information between IPsec gateways failover)

Can it be replicate on GNU/Linux systems? It's efficient and working very well.

Happy to discuss about it and if I can help to make this feature alive, I will gladly do it.

Regards,

**Comment by tobiasbrunner** [comment]
> It doesn't seem to work anymore on a vast majority of GNU/Linux distribution either using IPtables or nftables.

It never worked with distribution kernels because it requires a kernel patch to work properly, which now includes the CLUSTERIP module.

A more modern solution (supporting IPv6 and nftables) would be preferable, but unlikely to happen anytime soon.

**Reply by avachette** [reply]
> It never worked with distribution kernels because it requires a kernel patch to work properly, which now includes the CLUSTERIP module.

You mean even now?

The thing is that users wanting to do HA with Strongswan needs to compile a custom kernel, it's not an out of the box option.
 
> A more modern solution (supporting IPv6 and nftables) would be preferable, but unlikely to happen anytime soon.

What about the option I stated earlier like the one done on OpenBSD? Or do you see blockers to implement it that way on GNU/Linux distribution?

It's more about having pointers if I want to PoC around that kind of solution.

**Reply by tobiasbrunner** [reply]
> You mean even now?

Yes, now especially because CLUSTERIP was removed from the kernel with 6.3.
